3p/recaptcha.js

```diff
@@ -41,28 +47,110 @@ const TAG = 'RECAPTCHA';
 /** @const {string} */
 const RECAPTCHA_API_URL = 'https://www.google.com/recaptcha/api.js?render=';
 
+/** {?IframeMessaginClient} **/
+let iframeMessagingClient = null;
+
+/**
+ * Initialize 3p frame.
+ */
+function init() {
+  initLogConstructor();
+  setReportError(console.error.bind(console));
+}
+
+// Immediately call init
+init();
+
 /**
- * @param {!Window} global
- * @param {!Object} data
+ * Main function called by the recaptcha bootstrap frame
  */
-export function recaptcha(global, data) {
+function recaptcha() {
 
-  validateData(data, ['sitekey', 'action']);
+  /**
+   *  Get the data from our name attribute
+   * sitekey {string} - reCAPTCHA sitekey used to identify the site
+   * sentinel {string} - string used to psuedo-confirm that we are
+   *    receiving messages from the recaptcha frame
+   */
+  let dataObject;
+  try {
+    dataObject = parseJson(window.name);
+  } catch (e) {
+    dev().error(TAG + ' Could not parse the window name.');
```

Comment:
Yes I do, since domain verification is going to be very specific with the recaptcha, I think it is important that we try to handle this here, rather than when their API returns an error.